### DISCUSSION TITLE
Typescript allowing undefined when parameters should not be used

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#17072] [Q&A] [answered]
Hello, 

I am working with Nextjs 13, Typescript Postgresql and would like to leverage typesafety features. 

For simplicity, let's look at the following schema :  
```
model man{
 id       Int  @id @default(autoincrement())
 start?  Date    //start is optional
 name  @default("John") 
 }

```

And I build a create query like this ;  
```
export async function CreateMan( man_data: Man){
const query = await db.man.create({data :  man_data});
}
```
While this would work, I would like the flexibility to have undefined fields in the man_data object, so that some parameters are not defined, for example the id. 
Indeed, If man_data = { id : undefined, ...} that will not work for typescript as id can only be an int. 

Is there a solution for that ? for example a generated types somewhere that would allow undefined to all the fields that have @default  for example ? so that I could  man_data: ManWithUndefined    for example

Hope my issue is clear and you can help me :)

**Comment by laurent512** [comment]
After ignoring typescript verification on that file, the build on vercel failed with other errors that seems to indicate an issue with types:
` Type error: '"/vercel/path0/node_modules/.prisma/client/index".Prisma' has no exported member named 'RruleCreateInput'. Did you mean 'UserCreateInput'?`

**Reply by laurent512** [reply]
Please not that I don't have these issues with npm run dev locally

**Comment by laurent512** [comment, marked as answer]
Issue solved by adding vercel build in package.json .
described here : https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-vercel

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
